packages/plugin-ext/src/main/node/plugin-localization-backend-contribution.ts

```diff
@@ -0,0 +1,21 @@
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { LocalizationBackendContribution } from '@theia/core/lib/node/i18n/localization-backend-contribution';
+import { PluginDeployer } from '../../common/plugin-protocol';
+import { PluginDeployerImpl } from './plugin-deployer-impl';
+import { Deferred } from '@theia/core/src/common/promise-util';
+
+@injectable()
+export class PluginLocalizationBackendContribution extends LocalizationBackendContribution {
+    @inject(PluginDeployer)
+    protected readonly pluginDeployer: PluginDeployerImpl;
+
+    override async initialize(): Promise<void> {
+        const pluginsDeployed = new Deferred();
+        this.pluginDeployer.onDidDeploy(() => {
+            pluginsDeployed.resolve();
+        });
+
+        await Promise.all([this.localizationRegistry.initialize(), pluginsDeployed.promise]);
+        this.initialized.resolve();
+    }
```

Comment:
Suggestion: While this works quite well, the idea behind introducing `waitForInitialization` was that we would be able to hide the default behavior of `super.initialize()`:
```suggestion
    protected readonly pluginsDeployed = new Deferred();

    override async initialize(): Promise<void> {
        this.pluginDeployer.onDidDeploy(() => {
            this.pluginsDeployed.resolve();
        });
        await super.initialize();
    }
    
    override waitForInitialization(): Promise<void> {
       return Promise.all([super.waitForInitialization(), this.pluginsDeployed.promise]);
    }
```